### PR TITLE
Nbozic master transaction api

### PR DESCRIPTION
**Pull request by nbozic**
@looser990 multi-json is just library which is creating compatible API with many JSON libraries out there, so if you want to switch you have easy time. Oj library is one of those which has nice documentation, it is recommended as first choice because of speed and has option to recursively serialize deep objects.

However there is a problem, when doing serialization one of fields is ^o and value is object (check out the test).

So basically we have two options:

1. Leave it like this and use it, while sending one additional field (we should evaluate do we need library for this)
2. Remove both multi-json and oj and implement to_json on base entity, which will work and when you call to_json on parent object it will call to_json on all underlying objects in recursion

Check this out and decide which way we should go.

### CODE REVIEW

**Comment by nbozic** [comment]
@looser990 Solved it, overriden `to_json` so it works and used Oj compact which compatibility mode which is calling `to_json`. This way we can control `to_json` and Oj is nice library with many controlling options and good documentation. Check it out and merge if ok, and I will add other Transaction tests in separate PR.
